lib/prelude/src/lib.rs

```diff
@@ -467,3 +467,31 @@ impl KnownTypeValue for False {
         false
     }
 }
+
+/// Alias for `for<'t> &'t Self : Into<T>`.
+pub trait RefInto<T> = where for<'t> &'t Self : Into<T>;
+
+
+
+// =============
+// === Owned ===
+// =============
+
+pub trait AsOwned {
+    type Owned;
+}
+
+impl<T> AsOwned for &T {
+    type Owned = T;
+}
+
+pub type Owned<T> = <T as AsOwned>::Owned;
+
+pub trait IntoOwned = AsOwned + Into<Owned<Self>>;
+
+
+
+/// Placeholder type used to represent any value. It is useful to define type-level relations like
```

Comment:
maybe `to represent any value type`? I mean, after this sentence I expect some variant over everything.

Reply:
right